src/commands/AdventOfCodeCommand.ts

```diff
@@ -7,14 +7,15 @@ import {Discord, Slash, SlashOption} from "discordx";
 
 @Discord()
 class AdventOfCodeCommand {
-	@Slash("aoc", {description: "Advent Of Code"})
+	@Slash({ name: "aoc", description: "Advent Of Code" })
 	async onInteract(
-		@SlashOption("year", {type: "NUMBER", minValue: 2015, required: false}) year: number,
-		@SlashOption("name", {type: "STRING", required: false}) name: string,
-			interaction: CommandInteraction): Promise<void> {
+		@SlashOption({ name: "year", description: "AOC year", type: ApplicationCommandOptionType.Number, minValue: 2015, required: false }) year: number | undefined,
+		@SlashOption({ name: "name", description: "User's name", type: ApplicationCommandOptionType.String, required: false }) name: string | undefined,
```

Comment:
So that's how I had it originally, there's some other commands that have this as well (InspectCommand for instance). If i change it to `?: string` it becomes an optional parameter (not sure why it's so much different that adding an undefined) but if it's an optional parameter it has to come after the `interaction` parameter. 

Doing that throws an error when using the command, I assume the `CommandInteraction` needs to be the parameters in the Slash command so how I see it it's necessary to do it the way I ended up doing it. Using the option params in `InspectCommand` throws this error:

```js
D:\dev\CodeSupport\discord-bot\src\commands\InspectCommand.ts:14
                const userObj = await DiscordUtils.getGuildMember(userID === undefined ? interaction.user.id : userID.id, interaction.guild!);
                                                                                                                        ^
TypeError: Cannot read properties of undefined (reading 'guild')
    at InspectCommand.onInteract (D:\dev\CodeSupport\discord-bot\src\commands\InspectCommand.ts:14:121)
    at next (D:\dev\CodeSupport\discord-bot\node_modules\discordx\src\decorators\classes\Method.ts:88:63)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)```

As first parameter infront of the `SlashOption` the `interaction` is undefined when I log it

Reply:
Ah, interesting. Okay, let's leave it as it is then.